contracts/programs/gummyroll/src/state/mod.rs

```diff
@@ -65,10 +72,23 @@ impl<const MAX_DEPTH: usize> From<(Box<ChangeLog<MAX_DEPTH>>, Pubkey, u64)>
 #[derive(BorshDeserialize, BorshSerialize)]
 #[repr(C)]
 pub struct MerkleRollHeader {
+    /// Buffer of changelogs stored on-chain. Must be a power of 2;
+    /// Valid options limited to: `(8, 64, 256, 512, 1024, 2048)`
     pub max_buffer_size: u32,
+
+    /// Depth of the Merkle tree to store.
+    /// Valid options are any integer between 14-30
+    /// Tree capacity can be calculated as power(2, max_depth)
     pub max_depth: u32,
+
+    /// Authority that validates the content of the trees.
+    /// Typically a program, e.g., the Bubblegum contract validates that leaves are valid NFTs.
     pub authority: Pubkey,
+
+    /// Authority that is responsible for signing for new additions to the tree.
     pub append_authority: Pubkey,
+
+    /// TODO(jon): Not sure what this is!
```

Comment:
The idea is that we can use this information in the indexer to know how far back to look if we need to rebuild the tree from scratch